membership-common/src/main/scala/com/gu/memsub/subsv2/Subscription.scala

```diff
@@ -108,23 +84,21 @@ object GetCurrentPlans {
       just checks that the sub.startDate is before, or the same as, the date received by this function.
        */
       val ensureStarted = unvalidated.ensure(DiscardedPlan(plan, s"hasn't started as of $dateToCheck").wrapNel)(_)
-      val alreadyStarted = plan match {
-        case _: Contributor => ensureStarted(_ => sub.startDate <= date)
-        case _ => ensureStarted(_.start <= dateToCheck)
-      }
-      val freePlanCancelled = alreadyStarted.ensure(DiscardedPlan(plan, "has a free plan which has been cancelled").wrapNel)(_)
+      val alreadyStarted =
+        if (plan.product == Contribution)
+          ensureStarted(_ => sub.startDate <= date)
+        else
+          ensureStarted(_.start <= dateToCheck)
       val contributorPlanCancelled =
         alreadyStarted.ensure(DiscardedPlan(plan, "has a contributor plan which has been cancelled or removed").wrapNel)(_)
       val paidPlanEnded = alreadyStarted.ensure(DiscardedPlan(plan, "has a paid plan which has ended").wrapNel)(_)
       val digipackGiftEnded = alreadyStarted.ensure(DiscardedPlan(plan, "has a digipack gift plan which has ended").wrapNel)(_)
-      plan match {
-        case _: FreeSubscriptionPlan[_, _] => freePlanCancelled(_ => !sub.isCancelled)
-        case plan: PaidSubscriptionPlan[_, _] if plan.product == Product.Contribution =>
-          contributorPlanCancelled(_ => !sub.isCancelled && !plan.lastChangeType.contains("Remove"))
-        case plan: PaidSubscriptionPlan[_, _] if plan.product == Product.Digipack && plan.charges.billingPeriod == OneTimeChargeBillingPeriod =>
-          digipackGiftEnded(_ => sub.termEndDate >= dateToCheck)
-        case plan: PaidSubscriptionPlan[_, _] => paidPlanEnded(_ => plan.end >= dateToCheck)
-      }
+      if (plan.product == Product.Contribution)
+        contributorPlanCancelled(_ => !sub.isCancelled && !plan.lastChangeType.contains("Remove"))
+      else if (plan.product == Product.Digipack && plan.charges.billingPeriod == OneTimeChargeBillingPeriod)
+        digipackGiftEnded(_ => sub.termEndDate >= dateToCheck)
+      else
+        paidPlanEnded(_ => plan.end >= dateToCheck)
```

Comment:
now that it's all runtime checks we can use if statements instead of pattern match